### PR TITLE
Migrate MIME native functions to interop

### LINKED ISSUE #19898: Migrate MIME to interop
**Description:**

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by arukshani**
Fixes #19898

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by wggihan** [comment]
@arukshani Shall we move multiple JAVA classes into few utils classes accordingly? Now we don't need to have single JAVA class for each Ballerina function.

**Comment by wggihan** [comment]
@arukshani Shall we move all these java classes into few util classes accordingly? Now we don't need to maintain single java class for each Ballerina function.

**Comment by wggihan** [comment]
@arukshani Shall we move all these java classes into few util classes accordingly? Now we don't need to maintain single java class for each Ballerina function.

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        if (headerName == null || headerValue == null) {
+            return;
+        }
+        HttpHeaders httpHeaders;
+        if (entityObj.getNativeData(ENTITY_HEADERS) != null) {
+            httpHeaders = (HttpHeaders) entityObj.getNativeData(ENTITY_HEADERS);
+        } else {
+            httpHeaders = new DefaultHttpHeaders();
+            entityObj.addNativeData(ENTITY_HEADERS, httpHeaders);
+        }
```
Comment (anupama-pathirage) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java]: The above logic is duplicated in setHeader method as well. Shall we move it to a seperate method?

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * @since 1.1.0
+ */
+public class EntityHeaders {
+    public static void addHeader(ObjectValue entityObj, String headerName, String headerValue) {
+        if (headerName == null || headerValue == null) {
+            return;
+        }
+        HttpHeaders httpHeaders;
+        httpHeaders = getHeaders(entityObj);
+        httpHeaders.add(headerName, headerValue);
```
Comment (wggihan) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java]: getHeaders(entityObj).add(headerName, headerValue); ?

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        HttpHeaders httpHeaders = (HttpHeaders) entityObj.getNativeData(ENTITY_HEADERS);
+        if (httpHeaders != null && httpHeaders.get(headerName) != null && !httpHeaders.get(headerName).isEmpty()) {
+            return httpHeaders.get(headerName);
+        } else {
+            throw BallerinaErrors.createError(HEADER_NOT_FOUND, "Http header does not exist");
+        }
+    }
+
+    public static ArrayValue getHeaderNames(ObjectValue entityObj) {
+        ArrayValue stringArray = new ArrayValueImpl(new BArrayType(BTypes.typeHandle));
```
Comment (wggihan) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java]: IINM, There is a new constructor for creating these values. BValueCreator or similar. We should use those. @vinok88

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            httpHeaders = (HttpHeaders) entityObj.getNativeData(ENTITY_HEADERS);
+        } else {
+            httpHeaders = setEntityHeaders(entityObj);
+        }
+        return httpHeaders;
+    }
+
+    private static HttpHeaders setEntityHeaders(ObjectValue entityObj) {
+        HttpHeaders httpHeaders;
+        httpHeaders = new DefaultHttpHeaders();
```
Comment (wggihan) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java]: HttpHeaders  httpHeaders = new DefaultHttpHeaders();

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/MimeEntityBody.java [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+    public static Object getBodyPartsAsChannel(ObjectValue entityObj) {
+        try {
+            String contentType = getContentTypeWithParameters(entityObj);
+            if (isMultipart(contentType)) {
+                String boundaryValue = HeaderUtil.extractBoundaryParameter(contentType);
+                String multipartDataBoundary = boundaryValue != null ? boundaryValue : getNewMultipartDelimiter();
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                MultipartDataSource multipartDataSource = new MultipartDataSource(entityObj, multipartDataBoundary);
+                multipartDataSource.serialize(outputStream);
+                EntityBodyChannel entityBodyChannel = new EntityBodyChannel(new ByteArrayInputStream(
```
Comment (wggihan) [inline stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/MimeEntityBody.java]: Don't we need to close these input/output streams?

Comment (arukshani) [reply stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/MimeEntityBody.java]: Only the outputstreams need closing.

File: stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java (hunk 3 above)
Comment (vinok88) [reply stdlib/mime/src/main/java/org/ballerinalang/mime/nativeimpl/EntityHeaders.java]: Yes, We have deprecated the ArrayValueImpl class. Please use the BValueCreator.createXXX methods.

**Pull request merged**